app/routes/user/password.tsx

```diff
@@ -0,0 +1,210 @@
+
+  useEffect(() => {
+    if (formRef.current && fetcher.state === 'submitting') {
+      formRef.current.reset()
+    } else if (formRef && formRef.current) {
+      // this works around autofill on the old password field for all browsers
+      setOldPassword((formRef.current[0] as HTMLInputElement).value || '')
+    }
+  }, [fetcher.state])
+
+  return (
+    <>
+      <h1>Reset Password</h1>
+      <>
+        <fetcher.Form method="POST" ref={formRef}>
+          <Label htmlFor="oldPassword">Old Password</Label>
+          <TextInput
+            data-focus
+            className={errorMessage && !oldPassword ? 'usa-input--error' : ''}
+            name="oldPassword"
+            id="oldPassword"
+            type="password"
+            placeholder="Old Password"
+            onChange={(e) => {
+              setOldPassword(e.target.value)
+            }}
+          />
+          {errorMessage === 'NotAuthorizedException' && (
+            <span className="text-red">Invalid Password</span>
+          )}
+          {errorMessage === 'LimitExceededException' && (
+            <span className="text-red">
+              Attempts Exceeded. Please try again later.
+            </span>
+          )}
+          <Label htmlFor="newPassword">New Password</Label>
+          <InputGroup error={isNewPasswordTouched && !valid}>
+            <TextInput
+              name="newPassword"
+              id="newPassword"
+              className={valid ? 'usa-input--success' : ''}
+              type={visible ? 'text' : 'password'}
+              placeholder="New Password"
+              onChange={(e) => {
+                setNewPassword(e.target.value)
+                setIsNewPasswordTouched(true)
+              }}
+              value={newPassword}
+            />
+            <div className="usa-input-suffix" aria-hidden="true">
+              {visible ? (
+                <Icon.VisibilityOff
+                  className="margin-y-2"
+                  aria-label="Hide password"
+                  onClick={() => {
+                    setVisible(!visible)
+                  }}
+                />
+              ) : (
+                <Icon.Visibility
+                  className="margin-y-2"
+                  aria-label="Show password"
+                  onClick={() => {
+                    setVisible(!visible)
+                  }}
+                />
+              )}
+            </div>
+          </InputGroup>
+          <FormGroup>
+            <ButtonGroup>
+              <Button disabled={shouldDisableSubmit} type="submit">
+                Reset Password
+              </Button>
+              {fetcher.state !== 'idle' && (
+                <>
+                  <Spinner /> Saving...
+                </>
+              )}
+              {fetcher.state === 'idle' && fetcher.data === null && (
+                <>
+                  <Icon.Check color="green" /> Saved
+                </>
+              )}
+            </ButtonGroup>
```

Comment:
To fix this vertical space issue:

<img width="281" alt="Screenshot 2023-07-19 at 10 44 06" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/728407/e5116e29-6123-447d-8f83-a5e601796dc0">

```suggestion
              <Button disabled={shouldDisableSubmit} type="submit">
                Reset Password
              </Button>
              {fetcher.state !== 'idle' && (
                <span className="text-middle">
                  <Spinner /> Saving...
                </span>
              )}
              {fetcher.state === 'idle' && fetcher.data === null && (
                <span className="text-middle">
                  <Icon.Check color="green" /> Saved
                </span>
              )}
```